db/db_iter.cc

```diff
@@ -680,8 +673,15 @@ bool DBIter::ReverseToForward() {
   // If that's the case, seek iter_ to current key.
   if (!expect_total_order_inner_iter() || !iter_.Valid()) {
     IterKey last_key;
-    last_key.SetInternalKey(ParsedInternalKey(
-        saved_key_.GetUserKey(), kMaxSequenceNumber, kValueTypeForSeek));
+    ParsedInternalKey pikey(saved_key_.GetUserKey(), kMaxSequenceNumber,
+                            kValueTypeForSeek);
+    if (timestamp_size_ > 0) {
+      // TODO: pre-create kTsMax.
+      const std::string kTsMax(timestamp_size_,
+                               static_cast<unsigned char>(0xff));
```

Comment:
Yes, it's implicit. THe plan is to extend the interface of Comparator so that application enabling timestamp feature needs to specify the largest and smallest timestamps. We currently do not do it.